### PR TITLE
[RFR] Fix SelectArrayInput

### LINKED ISSUE #1359: AutoComplete on ReferenceArrayInput is not working
Select multiple items from Database

When the page renders, the view shows for 1 second, then, for some reason, the page renders again an then the page gets broken and became full white.

**Related code:**

```
<Create title="Crear notificación" {...this.props}>
    <SimpleForm>
        <TextInput source="title" label="Título"/>
	<TextInput source="message" label="Mensaje" />
	<ReferenceArrayInput source="recipients" reference="users" label="Usuarios" filterToQuery={searchText => ({ firstName: searchText })} allowEmpty>
	     <SelectArrayInput optionText={optionRenderer} optionValue="playerId"/>
	</ReferenceArrayInput>
    </SimpleForm>
</Create>
```

**Other information:**

The console shows this
```
Uncaught TypeError: Cannot read property 'type' of undefined
    at AutoComplete.js:389
    at Array.every (<anonymous>)
    at AutoComplete.render (AutoComplete.js:366)
    at finishClassComponent (react-dom.development.js:7873)
    at updateClassComponent (react-dom.development.js:7850)
    at beginWork (react-dom.development.js:8225)
    at performUnitOfWork (react-dom.development.js:10224)
    at workLoop (react-dom.development.js:10288)
    at HTMLUnknownElement.callCallback (react-dom.development.js:542)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:581)
    at invokeGuardedCallback (react-dom.development.js:438)
    at renderRoot (react-dom.development.js:10366)
    at performWorkOnRoot (react-dom.development.js:11014)
    at performWork (react-dom.development.js:10967)
    at requestWork (react-dom.development.js:10878)
    at scheduleWorkImpl (react-dom.development.js:10732)
    at scheduleWork (react-dom.development.js:10689)
    at Object.enqueueSetState (react-dom.development.js:6212)
    at Connect../node_modules/react/cjs/react.development.js.Component.setState (react.development.js:237)
    at Connect.onStateChange (connectAdvanced.js:205)
    at Object.notify (Subscription.js:26)
    at Subscription.notifyNestedSubs (Subscription.js:65)
    at Connect.onStateChange (connectAdvanced.js:202)
    at Object.notify (Subscription.js:26)
    at Subscription.notifyNestedSubs (Subscription.js:65)
    at Connect.onStateChange (connectAdvanced.js:202)
    at dispatch (createStore.js:173)
    at middleware.js:13
    at middleware.js:66
    at dispatch (applyMiddleware.js:35)
    at utils.js:250
    at proc.js:479
    at exec (scheduler.js:19)
    at flush (scheduler.js:60)
    at asap (scheduler.js:33)
    at runPutEffect (proc.js:476)
    at runEffect (proc.js:425)
    at next (proc.js:306)
    at currCb (proc.js:378)
    at <anonymous>
```
```
The above error occurred in the <AutoComplete> component:
    in AutoComplete (created by ChipInput)
    in div (created by ChipInput)
    in ChipInput (created by SelectArrayInput)
    in SelectArrayInput (created by getContext(SelectArrayInput))
    in getContext(SelectArrayInput) (at create.jsx:13)
    in ReferenceArrayInput (created by Connect(ReferenceArrayInput))
    in Connect(ReferenceArrayInput) (created by ConnectedField)
    in ConnectedField (created by Connect(ConnectedField))
    in Connect(ConnectedField) (created by Field)
    in Field (created by FormFieldComponent)
    in FormFieldComponent (created by FormField)
    in FormField (created by FormInput)
    in div (created by FormInput)
    in FormInput (created by SimpleForm)
    in div (created by SimpleForm)
    in form (created by SimpleForm)
    in SimpleForm (created by Form(SimpleForm))
    in Form(SimpleForm) (created by Connect(Form(SimpleForm)))
    in Connect(Form(SimpleForm)) (created by ReduxForm)
    in ReduxForm (created by Connect(ReduxForm))
    in Connect(ReduxForm) (at create.jsx:9)
    in div (created by Card)
    in div (created by Paper)
    in Paper (created by Card)
    in Card (created by Create)
    in div (created by Create)
    in Create (created by WithPermissionsFilteredChildren)
    in WithPermissionsFilteredChildren (created by getContext(WithPermissionsFilteredChildren))
    in getContext(WithPermissionsFilteredChildren) (created by getContext(getContext(WithPermissionsFilteredChildren)))
    in getContext(getContext(WithPermissionsFilteredChildren)) (created by Connect(getContext(getContext(WithPermissionsFilteredChildren))))
    in Connect(getContext(getContext(WithPermissionsFilteredChildren))) (at create.jsx:8)
    in NotificationsCreate (created by Route)
    in Restricted (created by Connect(Restricted))
    in Connect(Restricted) (created by Route)
    in Route (created by CrudRoute)
    in Switch (created by CrudRoute)
    in CrudRoute (created by Route)
    in Route (created by AdminRoutes)
    in Switch (created by AdminRoutes)
    in AdminRoutes (created by Connect(AdminRoutes))
    in Connect(AdminRoutes) (created by getContext(Connect(AdminRoutes)))
    in getContext(Connect(AdminRoutes)) (created by Layout)
    in div (created by Layout)
    in div (created by Layout)
    in div (created by Layout)
    in div (created by Layout)
    in MuiThemeProvider (created by Layout)
    in Layout (created by WithWidth)
    in EventListener (created by WithWidth)
    in WithWidth (created by Connect(WithWidth))
    in Connect(WithWidth) (created by Route)
    in Route (created by Admin)
    in Switch (created by Admin)
    in div (created by Admin)
    in Router (created by ConnectedRouter)
    in ConnectedRouter (created by Admin)
    in TranslationProvider (created by withContext(TranslationProvider))
    in withContext(TranslationProvider) (created by Connect(withContext(TranslationProvider)))
    in Connect(withContext(TranslationProvider)) (created by Admin)
    in Provider (created by Admin)
    in Admin (created by withContext(Admin))
    in withContext(Admin) (at App.jsx:34)
    in App (at index.js:7)
```
image

**Environment**

* Admin-on-rest version: 1.3.3
* Last version that did not exhibit the issue (if applicable):
* React version: 16.2.0
* Browser: Chrome 63.0.3239
* Stack trace (in case of a JS error):

### PR DESCRIPTION
**Pull request by djhi**
Fixes #1359

I confirm it now work on both react 15 and 16

### CODE REVIEW

File: src/mui/input/SelectArrayInput.js [hunk 1]
```diff
@@ -92,14 +88,12 @@ export class SelectArrayInput extends Component {
 
     handleAdd = newValue => {
         const values = [...this.state.values, newValue];
-        this.setState({ values });
-        this.handleChange(values);
+        this.setState({ values }, this.handleChange(values));
```
Comment (natrim) [inline src/mui/input/SelectArrayInput.js]: if i remember correctly then second param to setState should be callback
ie. `this.setState({ values }, () => this.handleChange(values));`

[the remark this answers is not in the document]
Comment (djhi) [reply src/mui/input/SelectArrayInput.js]: Thanks! Can you open an issue and/or a PR ? :)

[the remark this answers is not in the document]
Comment (natrim) [reply src/mui/input/SelectArrayInput.js]: i can but, the question is if to use
`this.setState({ values }, () => this.handleChange(values));`
or 
`this.setState({ values }, state => this.handleChange(state.values));`

the first one uses `values` set one line higher
the second one uses the one in actual state at triggering the function

so i have no idea if (and which) will break the fix or not

[the remark this answers is not in the document]
Comment (djhi) [reply src/mui/input/SelectArrayInput.js]: The second one should be used. That's actually the whole point of this callback: it ensure you are called with the updated state.

[the remark this answers is not in the document]
Comment (natrim) [reply src/mui/input/SelectArrayInput.js]: opened pull #1481

**Pull request merged**